lib/auth/join_github.go

```diff
@@ -46,8 +47,19 @@ func (a *Server) checkGitHubJoinRequest(ctx context.Context, req *types.Register
 		return nil, trace.BadParameter("github join method only supports ProvisionTokenV2, '%T' was provided", pt)
 	}
 
+	// enterpriseOverride is a hostname to use instead of github.com when
+	// validating tokens. This allows GHES instances to be connected.
+	enterpriseOverride := token.Spec.GitHub.EnterpriseServerHost
+	if enterpriseOverride != "" {
```

Comment:
I'm quite unfamiliar with this flow and I'm not sure what part of this feature is dedicated to enterprise flow. I would like to ask why  the licence `BuildEnterprise` check is done only for`if enterpriseOverride != ""` condition ?

Reply:
So, this field allows GitHub joining to be used against a GitHub Enterprise Server rather than GitHub. We determined that this should be an enterprise only feature.